doc/src/start_states.md

```diff
@@ -0,0 +1,118 @@
+The following explains the syntax and semantics of Start States in lrlex.<br>
+A working example can be found in the repository at [lrpar/examples/start_states](https://github.com/softdevteam/grmtools/tree/master/lrpar/examples/start_states)
+
+# Motivation
+
+Start states are a feature from lex which can be used to add a bit of context sensitivity to lexical analysis.
+For instance, they can be used to implement nested comments (see the example in the repository).  Such that
+the tokens start/end markers of tokens maintain balance.
+
+This is achieved by making rules which are qualified to match only when the lexer is in a particular state.
+Additionally the lexer has a stack of states, and matching rules perform actions which modify the stack.
+
+# The INITIAL start state 
+By default unless specified otherwise all lex rules are members of the *INITIAL* start state.
+
+```
+%%
+<INITIAL>a "A"
+<INITIAL>[\t \n]+ ;
+```
+
+Being equivalent to the lex file below with no start states specified.
+
+```
+%%
+a "A"
+[\t \n]+ ;
+```
+
+# Rules matching multiple states
+
+Rules can be matched in multiple states, just separate the states a rule should match in commas.
+The following matches the `a` character when in either of the states `FirstState` or `SecondState`.
+
+```
+<FirstState,SecondState>a "A"
+```
+
+# Adding a start state 
+Start stats come in two forms, *exclusive* and *inclusive*. These are given by `%x` and `%s` respectively.
+
+## Exclusive states
+In an exclusive state, the rule can be matched *only* if the rule begins with the state specified.
+In the following because `ExclState` is *exclusive*, the `#=` rule is only matched during the `INITIAL` state,
+while the `a` and `=#` characters are only matched while in the `ExclState`.
+
+```
+%x ExclState
+%%
+
+#= <+ExclState>
+<ExclState>a "A"
+<ExclState>=# <-ExclState>;
+```
+
+## Inclusive states
+
+Inclusive states are added to the set of rules to be matched when the start state is unspecified.
+
+```
+%s InclusiveState
+%%
+
+a "A"
+<InclusiveState>b "B"
+<INITIAL>#= <+InclusiveState>;
+<InclusiveState>=# <-InclusiveState>;
+```
+
+Is equivalent to the following using exclusive states.
+
+```
+%x Excl
+%%
+
+<INITIAL, Excl>a "A"
+<Excl>b "B"
+<INITIAL>#= <+Excl>;
+<Excl>=# <-Excl>;
+```
+
+# Start state operators
```

Comment:
I wonder if this section should come earlier and be called something like "Differences from POSiX lex"?

Reply:
I moved this one section higher, Which I believe does make sense because the section higher had start state operators in it, and the section renaming did seem appropriate too.

Should be fixed in [aaaa782](https://github.com/softdevteam/grmtools/pull/468/commits/aaaa78201d0a7158a8b5d76fb2e18f2c0d45e901)